### PR TITLE
Merge FluidConstants -> Master

### LINKED ISSUE #46: Fluid Constants aka Dynamic Variable Config
Production software for the variable robot parameters in the robot code, including:
- interface definitions
- when/how updated
- etc..
- tested independently
- merged and integrated with 2020 robot code in github
- ready for integration

### PR DESCRIPTION
**Pull request by nrgill28**
This adds FluidConstants - 'constant' variables that can be edited over network tables while the robot is disabled.

Closes #46:
Constants can be of any type, defined as `FluidConstant<type>`
Constants can be updated locally using `FluidConstant.setValue(T val)`
Constants can be updated remotely over network tables by changing the value there.
Visit the Wiki page in this repo to see more info
(NOTE: The robot must be disabled to update constants over network tables)

Tested using the standalone RoboRio and the WPILib 2020 robot code simulator.

### CODE REVIEW

**Review by Zinka010 — APPROVED** [review]
Excellent wiki and code!

**Review by Lockscout — APPROVED** [review]
Very descriptive documentation on Fluid 'constants'. Code looks fine.

File: .idea/uiDesigner.xml [hunk 1]
```diff
@@ -0,0 +1,124 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (georgetzavelas) [inline .idea/uiDesigner.xml]: What is this file for?

File: .idea/gradle.xml [hunk 2]
```diff
@@ -4,9 +4,11 @@
   <component name="GradleSettings">
     <option name="linkedExternalProjectsSettings">
       <GradleProjectSettings>
+        <option name="delegatedBuild" value="true" />
+        <option name="testRunner" value="GRADLE" />
         <option name="distributionType" value="DEFAULT_WRAPPED" />
         <option name="externalProjectPath" value="$PROJECT_DIR$" />
-        <option name="gradleJvm" value="12" />
+        <option name="gradleJvm" value="13" />
```
Comment (georgetzavelas) [inline .idea/gradle.xml]: Is this the new standard Java version that everyone should be using?

File: .idea/misc.xml [hunk 3]
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <project version="4">
   <component name="ExternalStorageConfigurationManager" enabled="true" />
-  <component name="ProjectRootManager" version="2" languageLevel="JDK_12" project-jdk-name="12" project-jdk-type="JavaSDK" />
+  <component name="ProjectRootManager" version="2" languageLevel="JDK_12" project-jdk-name="13" project-jdk-type="JavaSDK" />
```
Comment (georgetzavelas) [inline .idea/misc.xml]: Why is this `languageLevel="JDK_12"` but `project-jdk-name="13" `?

**Review by georgetzavelas — COMMENTED** [review]
Amazing work! Some questions for you

File: .idea/uiDesigner.xml (hunk 1 above)
Comment (nrgill28) [reply .idea/uiDesigner.xml]: It was auto-generated by IntelliJ so I just decided to leave it in. After a search though it seems I can probably remove it without issue

File: .idea/gradle.xml (hunk 2 above)
Comment (nrgill28) [reply .idea/gradle.xml]: I think WPILib actually recommends Java 11 instead of either 12 or 13. It is set to Java 12 on master, so I'll update it to 11

**Pull request merged**